sopel/modules/currency.py

```diff
@@ -4,43 +4,43 @@
 # Licensed under the Eiffel Forum License 2
 from __future__ import unicode_literals, absolute_import, print_function, division
 
-import time
 import re
 import requests
+import time
 
-from sopel.module import commands, example, NOLIMIT, rule
-from sopel.tools import stderr
 from sopel.config.types import StaticSection, ValidatedAttribute
+from sopel.logger import get_logger
+from sopel.module import commands, example, NOLIMIT, rule
 
-rates_fiat_json = {}
-rates_btc_json = {}
-
-
-fiat_url = 'https://api.exchangeratesapi.io/latest?base=EUR'
-fixer_url = 'http://data.fixer.io/api/latest?base=EUR&access_key={}'
-crypto_url = 'https://apiv2.bitcoinaverage.com/indices/global/ticker/short?crypto=BTC'
-regex = re.compile(r'''
+FIAT_URL = 'https://api.exchangeratesapi.io/latest?base=EUR'
+FIXER_URL = 'http://data.fixer.io/api/latest?base=EUR&access_key={}'
+CRYPTO_URL = 'https://apiv2.bitcoinaverage.com/indices/global/ticker/short?crypto=BTC'
+EXCHANGE_REGEX = re.compile(r'''
     ^(\d+(?:\.\d+)?)                                            # Decimal number
     \s*([a-zA-Z]{3})                                            # 3-letter currency code
     \s+(?:in|as|of|to)\s+                                       # preposition
     (([a-zA-Z]{3}$)|([a-zA-Z]{3})\s)+$                          # one or more 3-letter currency code
 ''', re.VERBOSE)
 
+LOGGER = get_logger(__name__)
+rates_fiat_json = {}
+rates_btc_json = {}
 
-def btc_rate(code, reverse=False):
-    global rates_btc_json
+# Config
+class CurrencySection(StaticSection):
+    fixer_io_key = ValidatedAttribute('fixer_io_key', default=None)
+    auto_convert = ValidatedAttribute('auto_convert', parse=bool, default=False)
 
-    search = 'BTC{}'.format(code)
 
-    if search in rates_btc_json:
-        rate = rates_btc_json[search]['averages']['day']
-    else:
-        return "Sorry {} isn't currently supported".format(code)
+def configure(config):
+    config.define_section('currency', CurrencySection, validate=False)
+    config.currency.configure_setting('fixer_io_key', 'API key for fixer IO. Leave blank to use exchangeratesapi.io:')
```

Comment:
This prompt line would read better if it was consistent about how it calls each site. They are `fixer.io` and `exchangeratesapi.io`, but one is called `fixer IO` instead.